docs/source/caching/query-watchers.mdx

```diff
@@ -1,35 +1,57 @@
 ---
-title: Query watchers
+title: Watching cached data
+description: Update your UI in response to cache changes
 ---
 
-Because the normalized cache deduplicates all your data, and if you defined your cache ids, you can use the cache as the source of truth for your UI and be automatically notified when some data changes. For this you can use `ApolloQueryCall.watch`
+Because your [normalized cache](./normalized-cache/) can deduplicate your GraphQL data (using proper [cache IDs](./declarative-ids/)), you can use the cache as the source of truth for populating your UI.
+
+When executing an operation, you can use the `ApolloQueryCall.watch` method to automatically notify that operation whenever its associated cached data changes:
+
+```kotlin{2}
+apolloClient.query(TodoDetailsQuery())
+  .watch()
+  .collect { response ->
+    // This code now executes on response *and*
+    // every time cached data changes
+  }
+```
+
+Let's look at some examples.
 
 ## Watching a query
 
-Assuming you have the following queries for a collaborative todo list app:
+Let's say we're building a collaborative todo list app that executes these two queries:
+
+import { CodeColumns } from 'gatsby-theme-apollo-docs';
+
+<CodeColumns>
 
 ```graphql
```

Comment:
```suggestion
```graphql:title=AllTodos.graphql
```

Reply:
Ahh so herein lies the current _issue_ with `CodeColumns`: codeblock titles are rendered weirdly, and they break the columns. But I'll put the filename in a comment!